code/modules/events/vent_clog.dm

```diff
@@ -0,0 +1,116 @@
+/datum/round_event_control/vent_clog
+	name = "Clogged Vents: Normal"
+	typepath = /datum/round_event/vent_clog
+	weight = 10
+	max_occurrences = 3
+	min_players = 10
+
+/datum/round_event/vent_clog
+	announceWhen	= 1
+	startWhen		= 5
+	endWhen			= 35
+	var/interval 	= 2
+	var/list/vents  = list()
+	var/randomProbability = 1
+	var/reagentsAmount = 500
+	var/list/saferChems = list(/datum/reagent/water,/datum/reagent/carbon,/datum/reagent/consumable/flour,/datum/reagent/space_cleaner,/datum/reagent/consumable/nutriment,/datum/reagent/consumable/condensedcapsaicin,/datum/reagent/drug/mushroomhallucinogen,/datum/reagent/lube,/datum/reagent/glitter/pink,/datum/reagent/cryptobiolin,
+						 /datum/reagent/toxin/plantbgone,/datum/reagent/blood,/datum/reagent/medicine/charcoal,/datum/reagent/drug/space_drugs,/datum/reagent/medicine/morphine,/datum/reagent/water/holywater,/datum/reagent/consumable/ethanol,/datum/reagent/consumable/cocoa/hot_cocoa,/datum/reagent/toxin/acid,/datum/reagent/toxin/mindbreaker,/datum/reagent/toxin/rotatium,/datum/reagent/bluespace,
+						 /datum/reagent/pax,/datum/reagent/consumable/laughter,/datum/reagent/concentrated_barbers_aid,/datum/reagent/colorful_reagent,/datum/reagent/peaceborg/confuse,/datum/reagent/peaceborg/tire,/datum/reagent/consumable/sodiumchloride,/datum/reagent/consumable/ethanol/beer,/datum/reagent/hair_dye,/datum/reagent/consumable/sugar,/datum/reagent/glitter/white,/datum/reagent/growthserum)
+	//needs to be chemid unit checked at some point
+
+/datum/round_event/vent_clog/announce()
+	priority_announce("The scrubbers network is experiencing a backpressure surge. Some ejection of contents may occur.", "Atmospherics alert", SSstation.announcer.get_rand_alert_sound())
+
+/datum/round_event/vent_clog/setup()
+	endWhen = rand(25, 100)
+	for(var/obj/machinery/atmospherics/components/unary/vent_scrubber/temp_vent in GLOB.machines)
```

Comment:
Isn't there a atmos_machinery global list? The one used by xenos to find vent spawns

Reply:
Same as above

Reply:
Seems I was wrong, GLOB.machines is exactly what xeno spawns use